<?xml version='1.0'?>
<launch>
	<!-- Navigator Node -->
	<node pkg="egh450g1_code" type="planner" name="planner" clear_params="true" output="screen">

		<!-- These topics need to be remapped correctly! -->
		<!-- Input to contrail for path tracking -->
		<param name="~contrail" value="/mavel/contrail" type='string'/>
		<!-- Service to request a safe path from breadcrumb -->
		<remap from="~request_path" to="/breadcrumb/request_path"/>

		<!-- Input to trigger diversion from imagery -->
		<remap from="~imagery_trigger" to="/imagery_trigger" />
		<!-- Input pose from imagery -->
		<remap from="~imagery_pose" to="/imagery_pose" />
		<!-- Nav status from nav script-->
		<remap from="~nav_status" to="/nav_status" />
		<!-- Pose state input of UAV -->
		<remap from="~pose" to="/vicon/uavusrg1/pose" />
	</node>

	<!-- Breadcrumb Node -->
	<node pkg="breadcrumb" type="breadcrumb_node" name="breadcrumb" clear_params="true" output="screen">
		<param name="search_heuristic" value="1" type="int"/>
		<param name="allow_diagonals" value="true" type="bool"/>
		<param name="obstacle_threshold" value="50" type="int"/>

		<!-- Occupancy grid input -->
		<!--remap from="~grid" to="/grid"/-->
		<remap from="~grid" to="/grid_expanded"/>
	</node>

	<!-- Grid Expansion Node -->
	<node pkg="egh450g1_code" type="expander" name="expander" clear_params="true" output="screen">
		<!-- Occupancy Grid Input -->
		<remap from="~grid_real" to="/grid/real" />
		<!-- Occupancy Grid Output -->
		<remap from="~grid_expanded" to="/grid/expanded"/>
	</node>
	
</launch>
